sensors/Bh1750.h

```diff
@@ -22,8 +22,20 @@ class Bh1750 : public Brightness {
       _present = _bh.begin(MODE);
   }
   void measure (__attribute__((unused)) bool async=false) {
-    if( present() == true ) {
-      _brightness = _bh.readLightLevel();
+    if( present() ) {
+      while (!_bh.measurementReady())

```

Comment:
I have a problem with the while loop. If things runs bad - the sensor never exists the measure method. This is not allowed.
I would also split the method - first part "prepare()" - which makes the test. Users can use it or not.

Reply:
I understand. Is it ok if it's converted to a loop which exits after a maximum time elapsed and break before if it reports ready?
With the prepare() method you mean to take out the auto calibration part into a separate function? How should a user choose to use it or not?

Reply:
May be make itswitchable would be enough. I do not use this sensor, but other may have problems with the changes. What about to add an additional template argument ADVANCED_MEASURE which default value false. So user can set it in the template to true and active the better measurement.

Reply:
I've updated the pull request. It checks for a define BH1750_AUTOCALIBRATE now. The endless loop now has a defined maximum of 360ms (which should be the maximum time according to the lib it can take).